subprojects/plugins/src/main/java/org/gradle/api/plugins/JvmTestSuitePlugin.java

```diff
@@ -104,7 +105,9 @@ private Configuration createTestDataVariant(Project project, JvmTestSuite suite,
         variant.setVisible(false);
         variant.setCanBeResolved(false);
         variant.setCanBeConsumed(true);
-        variant.extendsFrom(project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+        variant.extendsFrom(project.getConfigurations().getByName(suite.getSources().getImplementationConfigurationName()),
```

Comment:
Why does the test results directories variant need to extend from runtime onlys as well?

Reply:
This is for the case when `application -> direct -> transitive`, but a _test_ in `application` references a class of `transitive`.

I thought `runtimeClasspath` would've been sufficient here to cover everything, but apparently implementation + runtimeOnly is ideal.

Reply:
Makes sense.  I hadn't thought about that situation.